ons_alpha/topics/models.py

```diff
@@ -41,19 +44,91 @@ class TopicPage(BaseTopicPage):
     parent_page_types = ["topics.TopicSectionPage"]
     subpage_types = ["articles.ArticleSeriesPage", "bulletins.BulletinSeriesPage", "methodologies.MethodologyPage"]
     page_description = "A specific topic page. e.g. Public sector finance or Inflation and price indices"
+    summary = models.CharField(blank=True, max_length=255)
+    topic_page_nav = models.CharField(blank=True, max_length=255)
+        # a list of distinct page types for descendants and associated documents
+
+    content_panels = BasePage.content_panels + [
+        FieldPanel("summary"),
+        FieldPanel("topic_page_nav"),
+    ]
+
+    edit_handler = TabbedInterface(
+        [
+            ObjectList(content_panels, heading="Content"),
+            ObjectList(
+                [FieldPanel("topic")],
+                help_text="Select the topics that this page relates to.",
+                heading="Taxonomy",
+            ),
+            ObjectList(BasePage.promote_panels, heading="Promote"),
+        ]
+    )
 
     def clean(self):
         super().clean()
         if TopicSectionPage.objects.filter(topic=self.topic).exists():
             raise ValidationError({"topic": "Topic Section Page with this topic already exists."})
 
 
+    def topic_list_child_page_types(self):
+        """
+        For a topic page will return a distinct list of child page types
+        """
+        list_items = self.get_children().live().public().specific()
+        display_names = []
+        for menu_item in list_items:
+            display_names.append(menu_item.specific.page_type_display_name )
+        return sorted(set(display_names))
+
+    def subpage_list(self):
+        child_page_types = self.topic_list_child_page_types()
+        display_names = {}
+        for child_page_type in child_page_types:
+            display_names[child_page_type] = []
+            subpage_list = []
+
+            for subpage in self.get_children().live().public().specific():
+                if subpage.specific.page_type_display_name == child_page_type:
+                    if len(subpage.get_children().live().public().specific()) > 0:
```

Comment:
Suggestion: There's a more efficient way to determine this. Calling `len` will load all results into a list, and call `len` on it. `.exists` does a very short query in the DB to determine if any rows match.

```suggestion
                    if subpage.get_children().live().public().specific().exists():
```